### DISCUSSION TITLE
Axios Header Configuration

### DISCUSSION THREAD

**Discussion by StefanNeuser** [inertiajs/inertia#955] [Ideas]
Hi InertiaJs Team,

we really like your work here, keep going!

At the beginning of 2021 i was already asking for a possibility to set axios header config for inertiajs. Your answer was, that you are looking forward to refactor inertiajs that it doesnt make use of axios anymore. But it still does.

Background, we are serving Shopify apps. This apps are running in a iframe where we are not able to use cookies anymore because of the iframe policies that comes by google etc… 

I attached some screenshots of the change that would add the possibility for adding axios header config’s that some apps runnung in a iframe must have.

Best regards from Germany (and Colombia at the moment) ,

Stefan

<img width="687" alt="Screenshot 2021-10-25 at 16 56 58" src="https://user-images.githubusercontent.com/2176503/138776907-893d6881-739d-4ae8-a082-8c0ffa277b7c.png">
<img width="747" alt="Screenshot 2021-10-25 at 16 45 44" src="https://user-images.githubusercontent.com/2176503/138776911-4bd09a0b-976e-4301-96f0-2e69fd75fdde.png">
<img width="541" alt="Screenshot 2021-10-25 at 16 44 21" src="https://user-images.githubusercontent.com/2176503/138776915-364a84f7-c909-45e2-aab9-f8edabe0f880.png">

**Comment by ajnsn** [comment]
Hey @StefanNeuser. 

Replacing Axios seems to be still the plan as you can see here on the [1.0.0 milestone](https://github.com/inertiajs/inertia/milestone/1).

**Reply by StefanNeuser** [reply]
Hi, thanks for sharing that information. Do you know, if there will be a (global)-setting for add headers like this?

**Reply by ajnsn** [reply]
Looking at the [draft PR](https://github.com/inertiajs/inertia/blob/f2a6fdce98cf95baf80ce5623d4f73875a6ec294/packages/inertia/src/inertia.js#L70), there's at least a `headers` option for the `visit` method - but the PR may be outdated.

I just also noticed that there's the same `headers` option at the [current `visit` implementation](https://github.com/inertiajs/inertia/blob/master/packages/inertia/src/router.ts#L203), so at least you should be able to pass additional headers when using `visit`:

```js
this.$inertia.visit('profile', {
    headers: {
        'X-Hello': 'World',
    },
});
```
Partial headers of the request:
```

X-Inertia: true
Content-Type: application/json
X-Inertia-Version: 601b5234d2c5fa746d8038495235e6b0
X-Requested-With: XMLHttpRequest
X-Hello: World
```

Still not global but maybe a solution?
Edit: It's also in the [documenation](https://inertiajs.com/manual-visits#custom-headers), did not know that!

**Reply by StefanNeuser** [reply]
Yes i know but this is not helpful in a iframe third party app context. I think, to add globally headers is a really basic feature. For example, if you don't want to use cookies

**Reply by StefanNeuser** [reply]
But maybe you have more skills than me and can help me putting the code from the screenshots in the latest version of inertia and inertia-react... i'm stocking on typescript at the moment :-(

**Reply by ajnsn** [reply]
You could follow the `initialPage` to add your "headers" param to `init` method use that in the React adapter. For the type you could duplicate the PageProps. Just a  rough theory though.

**Reply by StefanNeuser** [reply]
thanks but i give up... typescript make me sad

**Comment by patjakubik** [comment]
Hey @StefanNeuser did you and your team figured it out? We have same issue, looks like nothing happened on the Inertia side meanwhile.
